core/db/mongo/metadata.go

```diff
@@ -99,7 +100,9 @@ func (m *MongoClient) GetScheduleEvent(se *models.ScheduleEvent, q bson.M) error
 	var mse mongoScheduleEvent
 
 	err := col.Find(q).One(&mse)
-	if err != nil {
+	if err == mgo.ErrNotFound {
```

Comment:
Question -- Why do some of these have an "else if" null check but others don't?

Could this be streamlined by creating a core/db/mongo/utils.go containing the following function?

```
func errorMap(e error) error {
    if e == mgo.ErrNotFound {
        e = db.ErrNotFound
    }
   return e
}
```

The caller would then be:
```
err := col.Find(q).One(&mse)
if err != nil {
    return errorMap(err)
}
```